wx/tools/wxget_docs_demo.py

```diff
@@ -0,0 +1,175 @@
+
+import sys
+import os
+import subprocess
+import webbrowser
+import tarfile
+if sys.version_info >= (3,):
+    from urllib.error import HTTPError
+    import urllib.request as urllib2
+    import urllib.parse as urlparse
+    from urllib.request import pathname2url
+else:
+    import urllib2
+    from urllib2 import HTTPError
+    import urlparse
+    from urllib import pathname2url
+
+import wx
+
+import wxget
+
+APP = None
+WXVERSION = wx.version().split(' ')[0]
+MAJOR = WXVERSION.split('.')[0]
+if MAJOR != '4':
+    raise ValueError("wx Versions other than 4 not supported!")
+
+def endsure_wx_app():
+    """ Ensure that there is a wx.App instance."""
+    global APP
+    if APP is None and not wx.GetApp():
+        APP = wx.App()
+        APP.SetAppName("wxPython")
+    return (APP is not None)
+
+def get_paths_dict():
+    """ Get a dictionary of the required paths."""
+    global APP
+    endsure_wx_app()
+    sp = wx.StandardPaths.Get()
+    pathdict = {}
+    pathdict['TempDir'] = sp.GetTempDir()
+    pathdict['Cache'] = os.path.join(sp.GetUserLocalDataDir(), 'wxDocsDemoCache',
+                                     WXVERSION)
+    pathdict['Docs_URL'] = wxget.get_docs_demo_url(False)
+    #pathdict['wxDocs'] = os.path.join(sp.GetAppDocumentsDir(), 'wxDocs', WXVERSION)
+    pathdict['wxDocs'] = sp.GetAppDocumentsDir()
+    pathdict['Docs_Name'] = "wxPython-docs-%s" % WXVERSION
+    pathdict['Demo_URL'] = wxget.get_docs_demo_url(True)
+    #pathdict['wxDemo'] = os.path.join(sp.GetUserLocalDataDir(), 'wxDemo', WXVERSION)
+    pathdict['wxDemo'] = sp.GetUserLocalDataDir()
+    pathdict['Demo_Name'] = "wxPython-demo-%s" % WXVERSION
+    pathdict['Ext'] = 'tar.gz'
+    return pathdict
+
+def unpack_cached(cached, dest_dir):
+    """ Unpack from the cache."""
+    print('Unpack', cached, 'to', dest_dir)
+    tf = tarfile.open(cached, "r:*")
+    tf.extractall(dest_dir)
+    dest_dir = os.listdir(dest_dir)[0]
+    return dest_dir
+
+def get_item(final, url, cache, name, ext):
+    """ Get the item """
+    print('Looking for', name, 'at', final)
+    fullpath = os.path.join(final, name)
+    if os.path.exists(fullpath):  # Already exists
+        return fullpath
+
+    cached = os.path.join(cache, name)
+    cached = os.path.extsep.join([cached, ext])
+    print('Looking for cashed', cached)
+    if not os.path.exists(cached):  # No cached copy
+        cached = wxget.download_file(url, cache, True)
+
+    if os.path.exists(cached):  # We now have a cached copy
+        unpack_cached(cached, final)
+    else:
+        fullpath = None
+    return fullpath
+
+def report_error(err_text):
+    """ Report a problem."""
+    endsure_wx_app()
+    wx.MessageBox(err_text, caption='ERROR!',
+                  style=wx.OK|wx.CENTRE|wx.ICON_ERROR)
+
+def done(result=0):
+    """ Tidy up and exit."""
+    global APP
+    if APP:
+        print("Closing Launcher App!")  # Debug
+        APP.Destroy()
```

Comment:
`wx.App.Destroy` can sometimes be unsafe to call, however since you are never calling `MainLoop` then it is sufficient to simply exit normally without cleaning up the app object. If you run into situations where that is not enough then using `wx.Exit()` to force it would be a better way to go.